src/Generator/InvoiceGenerator.php

```diff
@@ -78,6 +79,9 @@ public function generateForOrder(OrderInterface $order, \DateTimeInterface $date
         /** @var ChannelInterface $channel */
         $channel = $order->getChannel();
 
+        $paymentState = $order->getPaymentState() === PaymentInterface::STATE_COMPLETED ?
+            InvoiceInterface::PAYMENT_STATE_COMPLETED : InvoiceInterface::PAYMENT_STATE_PENDING;
```

Comment:
Should we be forced to be aware of the names of the specific states here? **What if** the `createForData` would take optional last argument `bool $isPaid = false` (which would be `$order->getPaymentState() === PaymentInterface::STATE_COMPLETED` for us) and create an invoice with a proper state inside? What say you? 👑 ⚔️ 💀